bundles/mapping/mapwfs2/service/StatusHandler.js

```diff
@@ -1,118 +1,188 @@
+      if(data.message === 'started') {
+        this.__log('Processing started for layer ' + data.layerId + ' (req: ' + data.reqId + ')');
+        if(typeof this.timer === 'undefined'){
+          this.timer = setTimeout(function(){
+            this.__log('Processing layer ' + data.layerId + ' takes longer than excpected');
+          }, 4000);
+        }
+        return;
+      }
+      else if(data.message === 'completed') {
+        clearTimeout(this.timer);
+        if(data.success) {
+          this.__log('Processing finished for layer ' + data.layerId + ' (req: ' + data.reqId + ')');
+        }
+        else {
+          this.__log('Processing error for layer ' + data.layerId + ' (req: ' + data.reqId + ')');
+        }
+        this.__handleCompleted(data);
+      }
+    },
+    handleError : function(error, plugin) {
+      this.__log('Error on layer ' +  error.layerId + ':' + error.message);
+      var status = this.getLayerStatus(error.layerId);
+      status.error.push(error.message);
+      var requestBuilder = Oskari.requestBuilder('ShowMessageRequest');
+      var layer = this.getMapLayer(error.layerId);
 
-	});
+      if( this._errorLayer === null ) {
+        this._errorLayer = error;
+      } else {
+        for(var i = 0; i < this._errorLayers.length; i++) {
+          //same layer don't do anythin
+          if( this._errorLayers[i].errorlayer.layerId === error.layerId) {
+            return;
+          }
+        }
+      }
+      if( error.type === 'normal' && !error.success) {
+        if(requestBuilder){
+          var request = requestBuilder(layer._name+": could not load layer", 'error');
+          Oskari.getSandbox().request('system-message', request);
+        } else {
+          Oskari.log(this.getName(), "no system-message started");
+        }
+        this._errorLayer = error;
+        this._errorLayers.push({errorlayer:this._errorLayer});
+      }
+      if( error.level === 'warning' ) {
+        if(requestBuilder){
+          var request = requestBuilder(layer._name+" "+error.message, 'warning');
+          Oskari.getSandbox().request('system-message', request);
+        } else {
+          Oskari.log(this.getName(), "no system-message started");
+        }
+        this._errorLayer = error;
+        this._errorLayers.push({errorlayer:this._errorLayer});
+      }
+      if(error.key === 'layer_scale_out_of_range'){
+        this.updateScale(layer, error.minscale, error.maxscale, plugin);
+      }
+
+      var sb = this.sandbox;
+      var loadEvent = sb.getEventBuilder('WFSStatusChangedEvent')(error.layerId);
+      loadEvent.setStatus(loadEvent.status.error);
+      loadEvent.setRequestType(loadEvent.type.image);
+      sb.notifyAll(loadEvent);
+    },
+    updateScale: function(layer, minscale, maxscale, plugin){
+      var me = this;
+      layer.setMinScale(minscale);
+      layer.setMaxScale(maxscale);
+      var olLayer = plugin.getOLMapLayers(layer)
+      olLayer[0].minScale = minscale;
```

Comment:
I think this will work only on openlayers 2. Should call some method on the plugin to update the olLayer and implement that method on ol2 and ol3 plugins